toolkit/docs/building/dedicated_guides/build_packages.md

```diff
@@ -0,0 +1,165 @@
+
+The following document shows the correct way to build packages for CBL-Mariner. Follow the instructions if you want to build the `.spec` files present in the core repo.
+
+## **1. Install Prerequisites**
+
+Prepare your system by [installing the necessary prerequisites here](../prerequisites.md).
+
+## **2. Pick a Branch or Tag**
+
+Please read [Clone and Sync to Stable Commit](../building.md#clone-and-sync-to-stable-commit) for details on picking a branch to develop from.
+
+For working with packages, the `hybrid` approach is often the most powerful, but the `stable` branches are the easiest. **Be aware that the stable branches trail behind the development branches and this can result in merge conflicts when it is time to create a PR.**
+
+The `hybrid` option minimizes the chance of package conflicts, but runs the risk of seeing unresolvable build cycles appearing.
+
+> **Regardless of your choice**, once you have done the initial development work, you should rebase and run a build on the development branch corresponding to your stable branch before creating a PR. Please reference the [Contribution Guide](CONTRIBUTING.md) for branch descriptions and further PR instructions.
+
+**If you are working on a package found in the [toolchain manifests](../../resources/manifests/package/)**, you should work directly in the development branches. Please refer to [Rebuild the Toolchain](../building.md#rebuild-the-toolchain) for instructions on rebuilding the toolchain.
+
+## **3. Get a Toolchain**
+
+If you are using the `stable` or `hybrid` branch strategy, the toolchain will be handled automatically. Just invoke the `sudo make build-packages ...` commands as needed and the tooling will deal with the rest.
+
+If you are working on a development branch or getting ready for a PR, you will need to build a toolchain. Refer to [Rebuild the Toolchain](../building.md#rebuild-the-toolchain) for details. **Save your toolchain archive for later!**
+
+## **4. Add/Modify a Package**
+
+See [Working with Packages](https://github.com/microsoft/CBL-MarinerTutorials/blob/-/docs/packages/working_with_packages.md#tutorial-customize-your-image-with-unsupported-packages) from the [CBL-MarinerTutorials](https://github.com/microsoft/CBL-MarinerTutorials) repo for details on working with RPM packages. This will cover:
+
+* Working with source files
+* Calculating package signatures
+* Writing .spec files
+
+The core repo behaves slightly differently than the dedicated build environment described in [CBL-MarinerTutorials](https://github.com/microsoft/CBL-MarinerTutorials), so use the steps described below to build your package instead.
+
+## **5. Build the Package**
+
+### Setup
+
+```bash
+# Clone the CBL-Mariner repo
+git clone https://github.com/microsoft/CBL-Mariner.git
+cd CBL-Mariner
+
+# Sync to the latest stable build
+git checkout 2.0-stable
+cd ./toolkit
+sudo make clean QUICK_REBUILD=y
+```
+
+### Build a toolchain if using an unstable branch
```

Comment:
Seems like there's many overlaps with the build images doc. Maybe we can separate these first few steps in another doc?